core/src/test/java/org/apache/iceberg/TestMetadataTableFilters.java

```diff
@@ -252,6 +254,310 @@ public void testPlanTasks() {
     validateCombinedScanTasks(tasks, 0);
   }
 
+  @Test
+  public void testPartitionSpecEvolutionRemovalV1() {
```

Comment:
nit: some comments in this test refer to delete files which cannot be part of v1 tables